src/Raven.Server/Documents/Handlers/Processors/Batches/ClusterTransactionRequestProcessor.cs

```diff
@@ -22,6 +27,18 @@ public ClusterTransactionRequestProcessor(DatabaseRequestHandler requestHandler,
 
         protected override ClusterConfiguration GetClusterConfiguration() => RequestHandler.Database.Configuration.Cluster;
 
+        public override Task<Task> WaitForDatabaseCompletion(Task<Task> onDatabaseCompletionTask, long index, CancellationToken token)
+        {
+            // check index
+            var lastCompleted = Interlocked.Read(ref RequestHandler.Database.LastCompletedClusterTransactionIndex);
+            if (lastCompleted >= index)
+            {
+                return Task.FromResult(Task.CompletedTask);
```

Comment:
here we aren't waiting for the index task

Reply:
Here we need to check if we have to wait for index, and then wait for the indexes to complete as of the current etag.
Need to only account for indexes modified by the actions in the tx

Reply:
Better yet, move the index wait entirely to here instead of the tx command

Reply:
Done